codex-rs/core/src/command_safety/is_dangerous_command.rs

```diff
@@ -0,0 +1,104 @@
+use crate::bash::try_parse_bash;
+use crate::bash::try_parse_word_only_commands_sequence;
+
+const DANGEROUS_COMMANDS: &[&str] = &["git reset"];
+
+pub fn command_might_be_dangerous(command: &[String]) -> bool {
+    if contains_dangerous_command(command) {
+        return true;
+    }
+
+    // Support `bash -lc "<script>"` where the any part of the script might contain a dangerous command.
+    if let [bash, flag, script] = command
+        && bash == "bash"
+        && flag == "-lc"
+        && let Some(tree) = try_parse_bash(script)
+        && let Some(all_commands) = try_parse_word_only_commands_sequence(&tree, script)
+        && all_commands
+            .iter()
+            .any(|cmd| contains_dangerous_command(cmd))
```

Comment:
I also don't think that this would catch: `bash -lc "bash -lc \"dangerous_command\""`. Should this be recursive in any way? Are we okay with a bit of a leaky sieve here since we also have the sandbox as a backstop?